timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -14,14 +14,48 @@ static_assert(HILBERT_N_BITS == 8);
 static_assert(HILBERT_GRID_DIM <= HILBERT_MAX_GRID_DIM);
 
 // generate kv values from coordinates to be radix sorted allowing the selection of a subset of coordinates
+template <typename RealType>
 void __global__ k_coords_to_kv_gather(
```

Comment:
Had noticed that this was using doubles to setup the hilbert sort which has no noticeable impact on overall performance and switching to float reduced the kernel run time by about 40% on average (as reported by `nsys profile` stats).

Reply:
Removed template parameter in [71a506d](https://github.com/proteneer/timemachine/pull/1042/commits/71a506d54bc7e924e2a020789f8ae4db08712d3c)

Reply:
Added to test here: https://github.com/proteneer/timemachine/pull/1042/commits/8e90fff8e11b36c939eb3d076b91ed8b2c58acbc

We also have a test that verifies that the neighborlist returns the same ixns regardless of permutation here: https://github.com/proteneer/timemachine/blob/master/tests/test_nblist.py#L151